PayloadPkg/OsLoader/Linux.c

```diff
@@ -244,23 +244,23 @@ SetupBzImage (
   //
   // Get graphics data
   //
-  GuidHob = GetNextGuidHob (&gLoaderFrameBufferInfoGuid, GetHobListPtr());
+  GuidHob = GetFirstGuidHob (&gEfiGraphicsInfoHobGuid);
   if (GuidHob != NULL) {
-    FrameBufferInfo = (FRAME_BUFFER_INFO *)GET_GUID_HOB_DATA (GuidHob);
+    GfxInfoHob = (EFI_PEI_GRAPHICS_INFO_HOB *)GET_GUID_HOB_DATA (GuidHob);
     ZeroMem (&Bp->ScreenInfo, sizeof (Bp->ScreenInfo));
     Bp->ScreenInfo.OrigVideoIsVGA  = VIDEO_TYPE_EFI;
-    Bp->ScreenInfo.LfbLinelength   = FrameBufferInfo->BytesPerScanLine;
-    Bp->ScreenInfo.LfbDepth        = (UINT16)FrameBufferInfo->BitsPerPixel;
-    Bp->ScreenInfo.LfbBase         = (UINT32)FrameBufferInfo->LinearFrameBuffer;
-    Bp->ScreenInfo.LfbWidth        = (UINT16)FrameBufferInfo->HorizontalResolution;
-    Bp->ScreenInfo.LfbHeight       = (UINT16)FrameBufferInfo->VerticalResolution;
+    Bp->ScreenInfo.LfbLinelength   = (UINT16) (GfxInfoHob->GraphicsMode.PixelsPerScanLine * 4);
+    Bp->ScreenInfo.LfbDepth        = 32;
+    Bp->ScreenInfo.LfbBase         = (UINT32)(UINTN)GfxInfoHob->FrameBufferBase;
+    Bp->ScreenInfo.LfbWidth        = (UINT16)GfxInfoHob->GraphicsMode.HorizontalResolution;
+    Bp->ScreenInfo.LfbHeight       = (UINT16)GfxInfoHob->GraphicsMode.VerticalResolution;
     Bp->ScreenInfo.Pages           = 1;
     Bp->ScreenInfo.BlueSize        = 8;
-    Bp->ScreenInfo.BluePos         = FrameBufferInfo->Blue.Position;
+    Bp->ScreenInfo.BluePos         = 0x00;

```

Comment:
why do we hardcode ?   we should read it from EFI_PIXEL_BITMASK in HOB.

Reply:
Same for GreenPos and RedPos.

Reply:
All these hardcoded value copied from Stage2Support.c when it builds FB HOB. 
will try to remove hard code value when possible.

Reply:
ok